demos/tutorial.md

```diff
@@ -0,0 +1,390 @@
+{
+    switch (ev.type)
+    {
+    case EventType.key:
+        // Process event
+        switch (ev.key.key)
+        {
+        case Key.esc:
+            {
+                s.stop();
+                exit(0);
+            }
+            break;
+        default:
+            break;
+        }
+        break;
+    case EventType.resize:
+        s.resize();
+        s.sync();
+        break;
+    }
+}
+```
+
+## Demo application
+
+The following demonstrates how to initialize a screen, draw text/graphics and handle user input.
+
+```d
+import std.stdio;
+import std.string;
+import std.conv;
+
+import dcell;
+
+void order(ref int i1, ref int i2)
+{
+    if (i2 < i1)
+    {
+        int v = i1;
+        i1 = i2;
+        i2 = v;
+    }
+}
+
+void order(ref Coord c1, ref Coord c2)
+{
+    order(c1.x, c2.x);
+    order(c1.y, c2.y);
+}
+
+void drawText(Screen s, Coord c1, Coord c2, Style style, string text)
+{
+    auto row = c1.y;
+    auto col = c1.x;
+    auto width = c2.x - c1.x;
+    int printed = 0;
+    while (printed < text.length)
+    {
+        if (width >= text[printed .. $].length)
+        {
+            s.style = style;
+            s.position = Coord(col, row);
+            s.write(text[printed .. $]);
+            break;
+        }
+        else
+        {
+            s.style = style;
+            s.position = Coord(col, row);
+            s.write(text[printed .. printed + width]);
+            printed += width;
+            row++;
+            col = c1.x;
+            if (row > c2.y)
+                break;
+        }
+    }
+}
+
+void drawBox(Screen s, Coord c1, Coord c2, Style style, string text, dchar fill = ' ')
+{
+    order(c1, c2);
+
+    Coord pos;
+    for (pos.x = c1.x; pos.x <= c2.x; pos.x++)
+    {
+        s[Coord(pos.x, c1.y)] = Cell(Glyph.horizLine, style);
+        s[Coord(pos.x, c2.y)] = Cell(Glyph.horizLine, style);
+    }
+    for (pos.y = c1.y + 1; pos.y < c2.y; pos.y++)
+    {
+        s[Coord(c1.x, pos.y)] = Cell(Glyph.vertLine, style);
+        s[Coord(c2.x, pos.y)] = Cell(Glyph.vertLine, style);
+    }
+    if (c1.y != c2.y && c1.x != c2.x)
+    {
+        s[Coord(c1.x, c1.y)] = Cell(Glyph.upperLeftCorner, style);
+        s[Coord(c2.x, c1.y)] = Cell(Glyph.upperRightCorner, style);
+        s[Coord(c1.x, c2.y)] = Cell(Glyph.lowerLeftCorner, style);
+        s[Coord(c2.x, c2.y)] = Cell(Glyph.lowerRightCorner, style);
+    }
+    for (pos.y = c1.y + 1; pos.y < c2.y; pos.y++)
+    {
+        for (pos.x = c1.x + 1; pos.x < c2.x; pos.x++)
+        {
+            s[pos] = Cell(fill, style);
+        }
+    }
+
+    auto newc1 = Coord(c1.x + 1, c1.y + 1);
+    auto newc2 = Coord(c2.x - 1, c2.y - 1);
+    drawText(s, newc1, newc2, style, text);
+}
+
+void main()
+{
+    import core.stdc.stdlib : exit;
+
+    auto s = newScreen();
+    assert(s !is null);
+    scope (exit)
+    {
+        s.stop();
+    }
+
+    s.start();
+    s.showCursor(Cursor.hidden);
+    s.enableMouse(MouseEnable.all);
```

Comment:
I would rather write something like:
```d
// Prefer this unless you really need drag:
s.enableMouse(); // buttons | motion

// If you need drag, explain the trade-off:
// s.enableMouse(MouseEnable.all); // includes drag; may impair OS copy/paste
```
screen.d documents that enabling drag impairs OS-level copy/paste. The demo forces MouseEnable.all, which includes drag, so better not do that unless user really wants it.